core/services/job/spawner.go

```diff
@@ -57,6 +57,7 @@ type (
 	// TODO(spook): I can't wait for Go generics
 	Delegate interface {
 		JobType() Type
+		BeforeJobCreated(spec Job)
```

Comment:
nit: Add comment that this will be called only when job is created for first time